addons/calendar/models/calendar_event.py

```diff
@@ -600,11 +613,16 @@ def _split_recurrence(self, time_values):
 
         :return: detached events
         """
+        self.ensure_one()
         if not time_values:
             return self.browse()
-
-        previous_week_day_field = weekday_to_field(self._get_start_date().weekday())
-        self.write(time_values)
+        if self.follow_recurrence and self.recurrency:
+            previous_week_day_field = weekday_to_field(self._get_start_date().weekday())
+        else:
+            # When we try to change recurrence values of an event not following the recurrence, we get the parameters from
+            # the base_event
+            previous_week_day_field = weekday_to_field(self.recurrence_id.base_event_id._get_start_date().weekday())
+        self.write({**time_values})
```

Comment:
Why this mess? Seems like a very expensive no-op.

Reply:
Why this mess? I don't know, I've just backported some code from Odoo 15 and master to get a calendar module that works better than in Odoo 14 (because some clients complained about a lot of issues).
I agree that the code of calendar/microsoft_calendar modules is not so clean but the plan was not to rewrite/refactor this code but just to get something that works (and it took a while to do that and test a lot of scenarios ...).
For this specific line, it is not a no-op, it gets the weekday field name from a weekday index.

Reply:
I'm talking about creating a dict by unpacking a dict into a dict.

Why not just pass `time_values` directly?

Reply:
Haha yes, you're right, I didn't notice that!